contracts/zaps/ZapCrvERC20.sol

```diff
@@ -0,0 +1,339 @@
+        ICurveFi _stableSwap,
+        IERC20 _lp
+    ) public Authorizable() {
+        _trancheFactory = __trancheFactory;
+        _trancheBytecodeHash = __trancheBytecodeHash;
+        token1 = _token1;
+        token2 = _token2;
+        stableSwap = _stableSwap;
+        _token1.approve(address(stableSwap), type(uint256).max);
+        _token2.approve(address(stableSwap), type(uint256).max);
+        _lp.approve(address(stableSwap), type(uint256).max);
+    }
+
+    // Memory encoding of the permit data
+    struct PermitData {
+        IERC20Permit tokenContract;
+        address who;
+        uint256 amount;
+        uint256 expiration;
+        bytes32 r;
+        bytes32 s;
+        uint8 v;
+    }
+
+    /// @dev Takes the input permit calls and executes them
+    /// @param data The array which encodes the set of permit calls to make
+    modifier preApproval(PermitData[] memory data) {
+        // If permit calls are provided we make try to make them
+        if (data.length != 0) {
+            // We make permit calls for each indicated call
+            for (uint256 i = 0; i < data.length; i++) {
+                _permitCall(data[i]);
+            }
+        }
+        _;
+    }
+
+    /// @dev Makes permit calls indicated by a struct
+    /// @param data the struct which has the permit calldata
+    function _permitCall(PermitData memory data) internal {
+        // Make the permit call to the token in the data field using
+        // the fields provided.
+        // Security note - This fairly open call is safe because it cannot
+        // call 'transferFrom' or other sensitive methods despite the open
+        // scope. Do not make more general without security review.
+        data.tokenContract.permit(
+            msg.sender,
+            data.who,
+            data.amount,
+            data.expiration,
+            data.v,
+            data.r,
+            data.s
+        );
+    }
+
+    /// @notice This function takes token2 and converts it into yield and principal tokens
+    /// of a specified tranche.
+    /// @dev `_amount` must be provided as value with this function call.
+    /// @param _amount Amount of token1 to convert.
+    /// @param _expiration Expiration of the target tranche.
+    /// @param _position The wrapped position contract address.
+    /// @param _ptExpected The minimum amount of principal tokens to mint.
+    /// @return returns the minted amounts of principal and yield tokens (PT and YT)
+    function zapIn_token1(
+        uint256 _amount,
+        uint256 _expiration,
+        address _position,
+        uint256 _ptExpected
+    ) external payable reentrancyGuard notFrozen returns (uint256, uint256) {
+        require(_amount != 0, "0 amount");
+
+        ITranche tranche = _deriveTranche(address(_position), _expiration);
+
+        //add liquidity with no minimum mint
+        stableSwap.add_liquidity([_amount, 0], 0);
+
+        uint256 outAmount = want.balanceOf(address(this));
```

Comment:
I don't see `want` declared anywhere.  Should this be a contract level immutable var?

Reply:
I also find the declaration of `want` to be wanting.